### PR TITLE
Fix #9871: use toNestedPairs in provablyDisjoint

### LINKED ISSUE #9871: Both tuple syntax are not the same to the compiler
## Minimized code

```Scala
  type IsTypeInTuple[T, Tup <: Tuple] = Tup match {
    case EmptyTuple => false
    case T *: ts => true
    case _ *: ts => IsTypeInTuple[T, ts]
  }
  summon[(Int *: String *: EmptyTuple) =:= (Int, String)] //they are the same
  summon[IsTypeInTuple[String, Int *: String *: EmptyTuple] =:= true] //compiles
  summon[IsTypeInTuple[String, (Int, String)] =:= true] //doesn't compile
  summon[IsTypeInTuple[String, Tuple.Widen[Int *: String *: EmptyTuple]] =:= true] // doesn't compiles
```

## Expectation

the last two summon should compile, just like the other cases.

### PR DESCRIPTION
**Pull request by OlivierBlanvillain**

### CODE REVIEW

**Comment by bishabosha** [comment]
This has been stale for a long time, would we be able to rebase?

File: tests/pos/9871.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+object Test {
+  type IsTypeInTuple[T, Tup <: Tuple] = Tup match {
+    case EmptyTuple => false
+    case T *: ts => true
+    case _ *: ts => IsTypeInTuple[T, ts]
+  }
+  summon[(Int *: String *: EmptyTuple) =:= (Int, String)] //they are the same
+  summon[IsTypeInTuple[String, Int *: String *: EmptyTuple] =:= true] //compiles
+  summon[IsTypeInTuple[String, (Int, String)] =:= true] //doesn't compile
+  summon[IsTypeInTuple[String, compiletime.Widen[Int *: String *: EmptyTuple]] =:= true] // doesn't compiles
```
Comment (bishabosha) [inline tests/pos/9871.scala:10]: `compiletime.Widen` has been removed so the test is failing

**Comment by OlivierBlanvillain** [comment]
@bishabosha Thanks for the pointer, it's green now!

**Review by bishabosha — APPROVED** [review]
LGTM

**Pull request merged**